src/main/java/org/obiba/es/opal/support/ESIndexManager.java

```diff
@@ -304,11 +308,22 @@ private void createIndexWithMapping() {
     }
 
     private void updateIndexWithMapping() {
-      log.info("Updating index mapping [{}] for {}", getIndexName(), name);
+      String indexName = getIndexName();
+      log.info("Updating index mapping [{}] for {}", indexName, name);
+
+      // get only the index settings (cannot reset shards or replicate settings)
+      Settings settings =
+          Settings.settingsBuilder().put(esSearchService.getClient().settings().getByPrefix("index.")).build();
+
+      // make sure to reset the configs in case they have been changed
+      esSearchService.getClient().admin().indices().close(new CloseIndexRequest(indexName)).actionGet();
+      esSearchService.getClient().admin().indices().prepareUpdateSettings(indexName).setSettings(settings).execute().actionGet();
+      esSearchService.getClient().admin().indices().open(new OpenIndexRequest(indexName)).actionGet();
```

Comment:
Shouldn't it be done in createIndexMapping() instead? These settings are not supposed to change when a new table is indexed.

Reply:
It would be better, will update accordingly.

Reply:
Actually no, since the index does not get deleted and recreated (similar to Mica), this was the only place I could make sure the new setting were applied.